addon/services/activity-watcher.ts

```diff
@@ -0,0 +1,152 @@
+import Service from '@ember/service';
+import { inject as service } from '@ember/service';
+
+import { Observable } from '@upfluence/hyperevents/helpers/observable';
+import EventsService, { ResourceEvent, prefixPath } from '@upfluence/hyperevents/services/events-service';
+import ToastService, { ToastOptions } from '@upfluence/oss-components/services/toast';
+
+type NotificationEvent = {
+  resource: string;
+  payload: {
+    notification_type: string;
+    timestamp: number;
+    data: any;
+  };
+};
+
+type RenderedNotification = {
+  title: string;
+  message: string;
+  type: string;
+};
+
+function renderNotificationMessage(message: string): RenderedNotification {
+  return {
+    title: `<i class="toast-title__icon upf-icon upf-icon--messages"></i>`,
+    message: message,
+    type: 'info'
+  };
+}
+
+function renderNotificationErrorMessage(message: string): RenderedNotification {
+  return {
+    title: `<i class="fa fa-info-circle" aria-hidden="true"></i>`,
+    message: message,
+    type: 'error'
+  };
+}
+
+function renderNotificationMessageWithAvatar(avatarUrl: string, message: string): RenderedNotification {
```

Comment:
It's already private as they can't be reached from outside, and don't require accessing the service state that's why leaving it outside felt like the right thing to do.

Reply:
We're not doing things this way anywhere else (at least from what I recall).
We have some "unscoped" functions for utilities that are exported but in this case only the service uses them. I don't understand why you would put them outside ?

Reply:
because they don't need access to the state of the service. Doing like this makes it less tempting to add some unnecessary coupling to the service state (ie `this`), as it would probably be adding logic where it is not needed.

Reply:
We will have to rework this once we have the new toasts anyway to avoid double error icons for example 🤷